src/integrationTest/java/uk/gov/hmcts/darts/task/service/GenerateCaseDocumentProcessorIntTest.java

```diff
@@ -78,6 +80,11 @@ void testGenerateCaseDocument() {
 
         List<ExternalObjectDirectoryEntity> eodCaseDocument = eodRepository.findByCaseDocument(caseDocument);
         assertThat(eodCaseDocument.size()).isEqualTo(1);
+
+        caseRepository.findById(courtCase.getId()).ifPresent(c -> {
+            assertThat(c.isRetentionUpdated()).isTrue();
+            assertThat(c.getRetentionRetries()).isEqualTo(0);
+        });
```

Comment:
Should add a generic catch incase this id could not be found fail the test

Reply:
technically you have a non null on case so if case document can be found so can case but I can add it just to bomb proof it